invenio_files_rest/models.py

```diff
@@ -1111,6 +1121,112 @@ def relink_all(cls, old_file, new_file):
             ObjectVersion.query.filter_by(file_id=str(old_file.id)).update({
                 ObjectVersion.file_id: str(new_file.id)})
 
+    def get_tags(self):
+        """Get tags for object version as dictionary."""
+        return {t.key: t.value for t in self.tags}
+
+
+class ObjectVersionTag(db.Model):
+    """Model for storing tags associated to object versions.
+
+    This is useful to store extra technical information for an object version.
+    """
+
+    __tablename__ = 'files_objecttags'
+
+    object_version_bucket_id = db.Column(
```

Comment:
Since these tags relate to a specific object version you can remove the bucket and key and just link them via ``ObjectVersion.version_id``. Naming-wise I would go with ``version_id`` instead of ``object_version_id`` (similar to naming used in MultipartObject).